### DISCUSSION TITLE
Lookup and UDAF

### DISCUSSION THREAD

**Discussion by Harold7665** [atoti/atoti#603] [Q&A]
Hello, is it possible to perform a lookup on table in a UDAF?

**Comment by CamilleBlanco** [comment]
Hi,
In atoti, store lookups are directly available through table joins, as explained in this[ discussion](https://github.com/atoti/atoti/discussions/592). Is there a reason why you could not use a table join in our use case? 🙂

**Reply by Harold7665** [reply]
Hello,
There is. The column we would need for a join is not based on facts but on a tree representation.
The DB contains:
**TRADES (base)
Trade ID (PK), Book ID (FK)**
T1, B
T2, B
T3, C
**BOOKS
Book ID (PK), Level 1, Level 2**
B, A, B
C, A, C
**LIMITS
Book ID (PK), Goal**
A, 10000
B, 2000
C, 3000
Then the tree would be:
```
A > count 0
- B > count 2
- C > count 1
```
And I would like to see:
```
Tree | Count | Goal
A    |   0   |   10000
- B  |   2   |    2000
- C  |   1   |    3000
```
The column on which the mapping is done should contain A, B, C. However, there is no fact related to A so this is only possible via a lookup I think.
